cards/portfolio/addon/components/isolated.js

```diff
@@ -52,6 +62,71 @@ export default LiveIsolatedCard.extend({
     return currencyGroups;
   }),
 
+  init() {
+    this._super(arguments);
+
+    if (this.web3.isMetamask) {
+      this.getMetamaskWallet.perform();
+    }
+  },
+
+  getMetamaskWallet: task(function * () {
+    let accounts = yield this.web3.eth.getAccounts();
+    if (accounts.length) {
+      let metamaskWallet = yield this.store.findRecord('wallet', 'metamask-wallet');
+
+      this.set('metamaskWallet', metamaskWallet);
+
+
+      accounts.forEach(account => {
+        this.getAssetsForWallet.perform(account);
+      });
+
+      this.content.get('wallets').pushObject(this.get('metamaskWallet'));
+
+      this.content.save();
+    }
+  }),
+
+  getAssetsForWallet: task(function * (account) {
+    this.addAssetToWallet.perform(account, 'ether');
+
+    for (let token of erc20Tokens) {
+      let tokenName = token.symbol.toLowerCase();
+      let balanceOf;
+
+      try {
+        balanceOf = yield this.store.findRecord(`${tokenName}-token-balance-of`, account.toLowerCase());
+      } catch (e) {
+        return;
+      }
+
+      if (balanceOf) {
+        this.addAssetToWallet.perform(`${account}_${tokenName}-token`, tokenName);
+      }
+    }
+  }),
+
+  addAssetToWallet: task(function * (assetId, networkId) {
+    let asset;
+    let network = yield this.store.findRecord('network', networkId);
+
+    try {
+      asset = yield this.store.findRecord('asset', assetId);
+    } catch (e) {
+      asset = this.store.createRecord('asset', {
+        id: assetId,
+        network,
+        todaysRatesLookup: this.todaysRatesLookup
```

Comment:
this should be a server side computed that is returned after the record is saved. there should be no reason to assert this.